includes/collection/class-actors.php

```diff
@@ -389,4 +389,247 @@ public static function get_type_by_id( $user_id ) {
+	 *
+	 * @param int $post_id The ID of the WordPress Custom-Post-Type.
+	 *
+	 * @return int The number of errors.
+	 */
+	public static function count_errors( $post_id ) {
+		return \count( \get_post_meta( $post_id, '_activitypub_errors', false ) );
+	}
+
+	/**
+	 * Clear the errors for a Follower.
+	 *
+	 * @param int $post_id The ID of the WordPress Custom-Post-Type.
+	 *
+	 * @return bool True on success, false on failure.
+	 */
+	public static function clear_errors( $post_id ) {
+		return \delete_post_meta( $post_id, '_activitypub_errors' );
+	}
+
+	/**
+	 * Get all Followers that had errors.
+	 *
+	 * @param int $number Optional. The number of Followers to return. Default 20.
+	 *
+	 * @return \WP_Post[] The Term list of Followers.
+	 */
+	public static function get_faulty( $number = 20 ) {
+		$args = array(
+			'post_type'      => self::POST_TYPE,
+			'posts_per_page' => $number,
+			// phpcs:ignore WordPress.DB.SlowDBQuery.slow_db_query_meta_query
+			'meta_query'     => array(
+				'relation' => 'OR',
+				array(
+					'key'     => '_activitypub_errors',
+					'compare' => 'EXISTS',
+				),
+				array(
+					'key'     => '_activitypub_inbox',
+					'compare' => 'NOT EXISTS',
+				),
+				array(
+					'key'     => '_activitypub_inbox',
+					'value'   => '',
+					'compare' => '=',
+				),
+			),
+		);
+
+		$posts = new \WP_Query( $args );
+		return $posts->get_posts();
+	}
+
+	/**
+	 * Get all Followers that have not been updated for a given time.
+	 *
+	 * @param int $number     Optional. Limits the result. Default 50.
+	 * @param int $older_than Optional. The time in seconds. Default DAY_IN_SECONDS.
+	 *
+	 * @return \WP_Post[] The list of Actors.
+	 */
+	public static function get_outdated( $number = 50, $older_than = DAY_IN_SECONDS ) {
+		$args = array(
+			'post_type'      => self::POST_TYPE,
+			'posts_per_page' => $number,
+			'orderby'        => 'modified',
+			'order'          => 'ASC',
+			'post_status'    => 'any', // 'any' includes 'trash'.
+			'date_query'     => array(
+				array(
+					'column' => 'post_modified_gmt',
+					'before' => \gmdate( 'Y-m-d', \time() - $older_than ),
+				),
+			),
+		);
+
+		$posts = new \WP_Query( $args );
+		return $posts->get_posts();
+	}
+
+	/**
+	 * Convert a Custom-Post-Type input to an Activitypub\Model\Actor.
+	 *
+	 * @param int|\WP_Post $post The post object.
+	 *
+	 * @return Actor|\WP_Error The Actor object or WP_Error on failure.
+	 */
+	public static function get_actor( $post ) {
+		$post = \get_post( $post );
+
+		if ( ! $post ) {
+			return new \WP_Error(
+				'activitypub_actor_not_found',
+				\__( 'Actor not found', 'activitypub' ),
+				array( 'status' => 404 )
+			);
+		}
+
+		/* @var Actor $object Actor object. */
+		$object = Actor::init_from_json( $post->post_content );
+
+		if ( \is_wp_error( $object ) ) {
+			return $object;
+		}
+
+		$object->set_id( $post->guid );
+		$object->set_name( $post->post_title );
+		$object->set_summary( $post->post_excerpt );
```

Comment:
Are ID, name, and summary not already part of the actor json?

Reply:
yes, but that makes the post at least a bit readable (when debugging) and we would not have to always JSON decode the content for simple UIs.

Reply:
I guess my question is, aren't these things already set through `$object = Actor::init_from_json( $post->post_content );` a few lines earlier?

Reply:
Ah, thought this was a different place in the code... sorry you are right!